driver/src/contracts/base_contract.rs

```diff
@@ -22,26 +34,81 @@ impl BaseContract {
         let web3 = web3::Web3::new(transport);
 
         let json: serde_json::Value = serde_json::from_str(&contents)?;
-        let abi: String = json.get("abi").ok_or("No ABI for contract")?.to_string();
+        let abi_string = json.get("abi").ok_or("No ABI for contract")?.to_string();
+        let abi = abi_string.as_bytes();
 
         let decoded_address = hex::decode(&address[2..])?;
         let contract_address: Address = Address::from(&decoded_address[..]);
-        let contract = Contract::from_json(web3.eth(), contract_address, abi.as_bytes())?;
+        let contract = Contract::from_json(web3.eth(), contract_address, abi)?;
+
+        let network_id = env::var("NETWORK_ID")?.parse()?;
+        let private_key = H256::from_str(&env::var("PRIVATE_KEY")?).unwrap();
```

Comment:
Should this error not also be propagated in case it fails to parse the private key?

Reply:
Also, for symmetry with the previous line, it is possibble to write:
```
let private_key = env::var("PRIVATE_KEY")?.parse()?;
```
and let type inference do its magic since `str::parse` is implemented for all `FromStr`.